docs/how_to/apphooks.rst

```diff
@@ -8,288 +10,186 @@ this case, you can create a normal django CMS page without any content of its
+******************
+Apphook management
+******************
 
-    class MyNamespacedApphook(CMSApp):
-        name = _("My Namespaced Apphook")
-        app_name = "myapp_namespace"
-        _urls = ["myapp.urls"]
+Uninstalling an apphook with applied instances
+==============================================
 
-    apphook_pool.register(MyNamespacedApphook)
+If you remove an apphook class (in effect uninstalling it) from your system that still has instances applied to pages,
+django CMS tries to handle this as gracefully as possible:
 
+* Affected Pages still maintain a record of the applied apphook; if the apphook class is reinstated, it will work as
+  before.
+* The page list will show apphook indicators where appropriate.
+* The page will otherwise behave like a normal django CMS page, and display its placeholders in the usual way.
+* If you save the page's Advanced settings, the apphook will be removed.
 
-.. note::
-    If you do provide an ``app_name``, then you will need to also give the app
-    a unique namespace in the *Advanced settings* of the page. If you do not, and
-    no other instance of the app exists using it, then the 'default instance
-    namespace' will be automatically set for you. You can then either reverse
-    for the namespace(to target different apps) or the app_name (to target
-    links inside the same app).
 
-If you use app namespace you will need to give all your view ``context`` a ``current_app``::
+Management commands
+===================
 
-    from django.core.urlresolvers import resolve
-    from django.shortcuts import render
+You can clear uninstalled apphook instances using a CMS management command ``uninstall apphooks``; for example::
 
-    def my_view(request):
-        request.current_app = resolve(request.path_info).namespace
-        return render(request, "my_template.html")
+    manage.py cms uninstall apphooks MyApphook MyOtherApphook
 
-.. note::
-    You need to set the current_app explicitly in all your view contexts as Django does not allow
-    any other way of doing this.
+You can get a list of installed apphooks using the :ref:`cms-list-command`; in this case::
 
-You can reverse namespaced apps similarly and it "knows" in which app instance it is:
+    manage.py cms list apphooks
 
-.. code-block:: html+django
+See the :ref:`Management commands reference <management_commands>` for more information.
 
-    {% url myapp_namespace:app_main %}
 
-If you want to access the same URL but in a different language use the language
-template tag:
+*************
+Apphook menus
+*************
 
-.. code-block:: html+django
+An apphook can also automatically attach a menu to a page (just as it would be attached if you attached the menu
+to the page in its *Advanced settings*).
 
-    {% load i18n %}
-    {% language "de" %}
-        {% url myapp_namespace:app_main %}
-    {% endlanguage %}
+Menus can be added to an apphook using the ``_menus`` attribute. On the basis of the example above::
```

Comment:
If we've made this attribute private, we probably should no longer document it or at least discourage its use here.